core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2129,6 +2123,16 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  private def withWorkspaceContextAndPermissionsF[T](workspaceName: WorkspaceName, requiredAction: SamResourceAction)(op: (SlickWorkspaceContext) => Future[T]): Future[T] = {
+    for {
+      workspaceContext <- dataSource.inTransaction { dataAccess =>
+        withWorkspaceContext(workspaceName, dataAccess) { ctx => DBIO.successful(ctx) }
+      }
+      opResult <- requireAccessF(workspaceContext.workspace, requiredAction) { op(workspaceContext) }
+    } yield opResult
+  }
+  // TODO: once everything is switched over to the Future-ified version, look for remaining calls to requireAccess and remove those too
+
```

Comment:
compare this Future-ified version to the `withWorkspaceContextAndPermissions` just previous to it (which the github UI has probably hidden)